api/src/test/java/org/apache/iceberg/types/TestTypeUtil.java

```diff
@@ -24,38 +24,43 @@
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import java.util.Set;
+import java.util.stream.Stream;
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.relocated.com.google.common.collect.Sets;
-import org.apache.iceberg.types.Types.IntegerType;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
 public class TestTypeUtil {
-  @Test
-  public void testReassignIdsDuplicateColumns() {
+  private static Stream<Arguments> testTypes() {
+    return Stream.of(Arguments.of(Types.IntegerType.get()), Arguments.of(Types.VariantType.get()));
+  }
+
+  @ParameterizedTest
+  @MethodSource("testTypes")
+  public void testReassignIdsDuplicateColumns(Type testType) {
```

Comment:
When updating existing tests, it is better to leave tests alone when they are testing behavior that is not relevant to what you're changing. In this case, the behavior being tested is not related to the type of these columns, it is that the columns duplicate one another.

It is typically a red flag for me to see updates to existing tests like this. When you're adding new behavior, think about what needs to be tested. For variant, this needs to test that variant columns are preserved (not removed, skipped, or replaced with `null`) and are reassigned new IDs. It does not need to test _how_ the IDs are assigned or that ID assignment is case sensitive.

Reply:
I was trying to add the coverage for Variant changes for the visitors such as ReassignIds or PruneColumns with the existing tests. I guess what you suggest with some additional tests should be able to provide coverage for them. Let me look into that.